src/py21cmfast/drivers/param_config.py

```diff
@@ -246,46 +345,21 @@ def from_inputstructs(
             astro_params=astro_params,
             flag_options=flag_options,
             random_seed=random_seed,
+            **kwargs,
         )
 
     @classmethod
-    def from_defaults(cls, random_seed, **kwargs):
-        """Construct full InputParameters instance from default values."""
-        cosmo_params = CosmoParams.new(
-            {
-                k: v
-                for k, v in kwargs.items()
-                if k in [f.name for f in attrs.fields(CosmoParams)]
-            }
-        )
-        user_params = UserParams.new(
-            {
-                k: v
-                for k, v in kwargs.items()
-                if k in [f.name for f in attrs.fields(UserParams)]
-            }
-        )
-        astro_params = AstroParams.new(
-            {
-                k: v
-                for k, v in kwargs.items()
-                if k in [f.name for f in attrs.fields(AstroParams)]
-            }
-        )
-        flag_options = FlagOptions.new(
-            {
-                k: v
-                for k, v in kwargs.items()
-                if k in [f.name for f in attrs.fields(FlagOptions)]
-            }
-        )
+    def from_defaults(cls, **kwargs):
```

Comment:
just wanted to check if this is still necessary, since we can just give the actual fields defaults now?

Reply:
I suppose not, there is a slightly different behaviour when someone tries to define an incomplete structure (via `from_output_structs` using an `InitialConditions` or `__init__` missing some arguments). Currently it refuses to let you build the structure at all, and you have to explicitly set the default structure if you want it. If we set default `InputStruct` fields here it will fill those in with the defaults, which may be more convenient (i.e the only required argument is the random seed). The only possible downside is that bad usage will be caught later: i.e ```InputParameters.from_output_struct((perturbed_field),...) --> ionize_box(astro_params=...)``` will catch errors during the ionize step instead of the parameter instantiation.